webapp/app/forms/flows/step_chooser.py

```diff
@@ -47,6 +48,9 @@ def get_correct_step(self, step_name) -> SteuerlotseStep:
             return RedirectSteuerlotseStep(self._get_possible_redirect(step_name), endpoint=self.endpoint)
         stored_data = get_session_data(self.session_data_identifier, default_data=self.default_data())
 
+        if request.method == 'POST':
```

Comment:
Mmmh, I'm not keen on introducing a dependency from the StepChooser to Flask / HTTP request processing. Maybe we can add a param `update_step_data` which the request handler in `routes.py` can set depending on the HTTP method?

Reply:
(On that note: I'm also not a fan of the step chooser loading session data, but that's another story… Let's actually do that architecture sketching afternoon we've been talking about once Julian is back!)